File: packages/volume/android/src/main/java/io/capawesome/capacitorjs/plugins/volume/classes/results/GetVolumeResult.java
package io.capawesome.capacitorjs.plugins.volume.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.volume.interfaces.Result;

public class GetVolumeResult implements Result {

    private final double volume;

    public GetVolumeResult(double volume) {
        this.volume = volume;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("volume", volume);
        return result;
    }
}

File: packages/volume/ios/Plugin/Classes/Results/GetVolumeResult.swift
import Foundation
import Capacitor

@objc public class GetVolumeResult: NSObject, Result {
    let volume: Float

    init(volume: Float) {
        self.volume = volume
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["volume"] = volume
        return result as AnyObject
    }
}
